benchexec/test_cgroups_pytest.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import subprocess
+import pytest
+
+from benchexec import check_cgroups
+
+logging.disable(logging.CRITICAL)
```

Comment:
This seems like it might change behavior compared to the previous code: So far it was executed right before the tests in class `TestCheckCgroups`, now it is executed at import time. Does this make a difference, e.g., if tests from more than one module are run and they all change the log level? Or does pytest ensure that the module is imported right before it is used or that such changes are otherwise isolated?

It seems that pytest has some [logging-specific features](https://docs.pytest.org/en/7.1.x/how-to/logging.html), maybe it is even possible to improve our log handling in tests compared to the existing situation?

Reply:
In pytest, each test function is run in a separate isolated environment, which means that changes to global state, such as log levels, made by one test function do not affect other test functions. So if tests from more than one module are run and they all change the log level, these changes will not affect each other.

However, if the log level is changed in a fixture that is shared across multiple tests, then the change to the log level may affect all the tests that use that fixture. In such cases, it's important to ensure that the fixture is designed to handle such global state changes in a way that is appropriate for the specific use case.

Therefore, if we change the log level in a module that has already been imported, those changes will persist across subsequent test runs unless you reset the log level manually. To avoid this, we can either reset the log level at the end of each test, or you can import the module inside a fixture that is scoped to the test function, so that the module is re-imported and its code is re-executed for each test.

Reply:
So, does this mean that we need a fixture here?

Actually, it would of course be nice if we do get the logging output if we want to debug the test. But it should not clutter the usual output. So far the workaround was to disable logging in the tests. But if we can improve this that would be great!